src/common/meta/src/rpc/router.rs

```diff
@@ -215,9 +208,10 @@ impl TableRoute {
     }
 
     pub fn find_region_leader(&self, region_number: RegionNumber) -> Option<&Peer> {
-        self.region_leaders
-            .get(&region_number)
-            .and_then(|x| x.as_ref())
+        self.region_routes
```

Comment:
I remember `region_leaders` has some caching purpose, to find region leader quickly. Is it has to be removed?